core/src/main/java/io/dekorate/SessionReader.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2018 The original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.dekorate;
+
+public interface SessionReader {
+
+  void read(Session session, Resources resources);
```

Comment:
There is no need to pass both Session and Resources, as one can grab the resources from the Session. See `SessionWritter`.